integration-templates/quickbooks/actions/create-invoice.ts

```diff
@@ -0,0 +1,73 @@
+import type { NangoAction, CreateInvoice, Invoice, ProxyConfiguration } from '../../models';
+import { getCompany } from '../utils/getCompany.js';
+import { toQuickBooksInvoice, toInvoice } from '../mappers/toInvoice.js';
+
+/**
+ * This function handles the creation of an invoice in QuickBooks via the Nango action.
+ * It validates the input invoice data, maps it to the appropriate QuickBooks invoice structure,
+ * and sends a request to create the invoice in the QuickBooks API.
+ * For detailed endpoint documentation, refer to:
+ * https://developer.intuit.com/app/developer/qbo/docs/api/accounting/all-entities/invoice#create-an-invoice
+ *
+ * @param {NangoAction} nango - The Nango action instance to handle API requests.
+ * @param {CreateInvoice} input - The invoice data input that will be sent to QuickBooks.
+ * @throws {nango.ActionError} - Throws an error if the input is missing or lacks required fields.
+ * @returns {Promise<Invoice>} - Returns the created invoice object from QuickBooks.
+ */
+export default async function runAction(nango: NangoAction, input: CreateInvoice): Promise<Invoice> {
+    const companyId = await getCompany(nango);
+
+    // Validate if input is present
+    if (!input) {
+        throw new nango.ActionError({
+            message: `Input invoice object is required. Received: ${JSON.stringify(input)}`
+        });
+    }
+
+    // Validate required fields
+    if (!input.customer_ref || !input.customer_ref.value) {
```

Comment:
Just a matter of thought we can generate zod models, maybe it's shorter and more reliable than manual validation

Reply:
I was thinking the same thing. Now that we also generate tests, it would be good practice to generate the Zod models and override them where possible. It would also be nice to apply this approach to the other models. What do you think?